metpy/calc/tools.py

```diff
@@ -1375,3 +1375,132 @@ def _abbrieviate_direction(ext_dir_str):
             .replace('SOUTH', 'S')
             .replace('WEST', 'W')
             )
+
+@exporter.export
+@preprocess_xarray
+def area_average(var,rad,lon,lat):
+    r"""Performs horizontal area-averaging of a field in latitude/longitude format over a specified radius.
+    Parameters
+    ----------
+    var : (M, N) ndarray
+        Variable to perform area averaging on. Can be 2, 3 or 4 dimensions. If 2D, coordinates must
+        be lat/lon. If using additional dimensions, area-averaging will only be performed on the last
+        2 dimensions, assuming those are latitude and longitude.
+    rad : `pint.Quantity`
+        The radius over which to perform the spatial area-averaging.
+    lon : array-like
+        Array of longitudes defining the grid
+    lat : array-like
+        Array of latitudes defining the grid
+        
+    Returns
+    -------
+    (M, N) ndarray
+        Area-averaged quantity, returned in the same dimensions as passed.
+    
+    Notes
+    -----
+    This function was originally provided by Matthew Janiga and Philippe Papin using a Fortran wrapper for NCL,
+    and converted to python with further efficiency modifications by Tomer Burg, with permission from the original
+    authors.
+    
+    This function assumes that the last 2 dimensions of var are ordered as (....,lat,lon). This can perhaps be modified
+    to handle (...,lon,lat) as well.
+    """
+    
+    #Inner function to perform the area-averaging
+    def calcavg(x,xavg,lon2d,lat2d,nlon,nlat,rad,box,eqrm):
+        nbox = (2*box+1)*(2*box+1)
+
+        #Iterate over latitude and longitude
+        for j in range((box),(nlon-box)):
+            for i in range((box),(nlat-box)):
+
+                lon1d = lon2d[i-box:i+box+1,j-box:j+box+1].reshape((nbox))
+                lat1d = lat2d[i-box:i+box+1,j-box:j+box+1].reshape((nbox))
+                x1d = x[i-box:i+box+1,j-box:j+box+1].reshape((nbox))
+
+                d1d = eqrm * np.sqrt(( (lon2d[i,j]-lon1d)*np.cos( (lat2d[i,j]+lat1d)/2.0 ) )**2.0 + (lat2d[i,j]-lat1d)**2.0)
+                z = x1d[d1d < rad] / len(x1d[d1d < rad])
+                xavg[i,j] = np.sum(z)
+
+        return xavg
+    
+    #convert passed radius to kilometers
```

Comment:
Would this work for a degree based radius or should we add logic/error handling there?